src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/ActivitySourceConfiguration.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    public sealed class ActivitySourceConfiguration : MonitoringSourceConfiguration
+    {
+        private readonly double _SamplingRatio;
+        private readonly string[] _ActivitySourceNames;
+
+        public ActivitySourceConfiguration(
+            DiagnosticsClient client,
+            double samplingRatio,
+            IEnumerable<string>? activitySourceNames)
+        {
+            _SamplingRatio = samplingRatio;
+            _ActivitySourceNames = activitySourceNames?.ToArray() ?? Array.Empty<string>();
+
+            if (_SamplingRatio < 1D)
+            {
```

Comment:
I don't think we should be doing something this heavyweight and async in the ctor. If we want to validate this only works in version 9, we can fail the pipeline later or have the validation done upfront in a helper async method.

Reply:
I moved it to a spot that felt more natural. Check it out, LMK.